core/dbt/cli/main.py

```diff
@@ -436,10 +444,21 @@ def source(ctx, **kwargs):
 @p.threads
 @p.vars
 @requires.preflight
+@requires.profile
+@requires.project
 def freshness(ctx, **kwargs):
     """Snapshots the current freshness of the project's sources"""
```

Comment:
[Not needed for this PR] Let's definitely remove the word "snapshots" from this help text, when we get to DX review (#6546). We renamed this command for a reason, to disambiguate it from `dbt snapshot`.